### PR TITLE
VRVU-17-2: add required fileds

### PR DESCRIPTION
**Pull request by jdolkowski**
https://sd-vxnaid.atlassian.net/browse/VRVU-17

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -41,9 +41,9 @@ android {
     signingConfigs {
         release {
             storeFile file("../support/keystore.jks")
-            storePassword SOLIDARITY_STORE_PASSWORD
+            storePassword "SOLIDARITY_STORE_PASSWORD"
             keyAlias 'releaseKey'
-            keyPassword SOLIDARITY_KEY_PASSWORD
+            keyPassword "SOLIDARITY_KEY_PASSWORD"
```
Comment (sherrif10) [inline app/build.gradle:46]: Well done on this great Pr. We're you supposed to commit this change ?

File: app/src/main/java/com/jnj/vaccinetracker/common/data/database/entities/ParticipantEntity.kt [hunk 2]
```diff
@@ -11,11 +11,14 @@ data class ParticipantEntity(
     override val participantUuid: String,
     @ColumnInfo(collate = ColumnInfo.NOCASE)
     override val participantId: String,
+    override val nin: String?,
```
Comment (sherrif10) [inline app/src/main/java/com/jnj/vaccinetracker/common/data/database/entities/ParticipantEntity.kt:14]: We need to add Nin validation to ensure a user is entering a right Nin. I have a pr that you might need to look into. Well it's not yet finished but you can glance at it and guide me accordingly. Thanks

File: app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt [hunk 3]
```diff
@@ -86,7 +87,8 @@ class DraftParticipantScript {
         val pictureResourceName: String? = "test_script_image.jpeg"
         val image = pictureResourceName?.let { readPicture(it) }
         val telephone: String? = null
-        val yearOfBirth = 1994
+        val birthDate = DateTime(1999, 6, 6)
+        val isBirthDateAnApproximation = false
```
Comment (sherrif10) [inline app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt:91]: How about we change the naming to **_isBirthDateEstimated_**

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            isApproximationChecked = checkBox.isChecked
+
+            findParent<BirthDatePickerListener>()?.onBirthDatePicked(selectedDate!!, isApproximationChecked)
+            dialog.dismiss()
+        }
+
+        dialog.findViewById<Button>(R.id.btn_cancel).setOnClickListener {
+            dialog.dismiss()
+        }
+
```
Comment (sherrif10) [inline app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt:52]: This looks very interesting. one small thing is to remove spaces that is not neccessary.

File: app/src/main/java/com/jnj/vaccinetracker/common/data/database/entities/ParticipantEntity.kt (hunk 2 above)
Comment (pwargulak) [reply app/src/main/java/com/jnj/vaccinetracker/common/data/database/entities/ParticipantEntity.kt]: @sherrif10 Let's merge this one, then rebase yours and merge yours.

File: app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt (hunk 3 above)
Comment (pwargulak) [reply app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt:91]: I agree, the 'Estimated' is the word used in OMRS in this case

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import android.widget.DatePicker
+import androidx.fragment.app.DialogFragment
+import com.jnj.vaccinetracker.R
+import com.jnj.vaccinetracker.common.helpers.findParent
+import com.soywiz.klock.DateTime
+import java.util.Calendar
+import androidx.appcompat.app.AlertDialog
+
+
+class BirthDatePickerFragment(
```
[the remark this answers is not in the document]
Comment (pwargulak) [reply app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt:16]: Hmm, I'm not sure of it. This is Date Picker component, class derives from DialogFragment.
I see that other classes in the package use pattern: <Funcitionality>Dialog

**Comment by pwargulak** [comment]
@jdolkowski Check our comments and I'll merge it

**Review by pwargulak — CHANGES_REQUESTED** [review]
Check comments

File: app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt (hunk 3 above)
Comment (jdolkowski) [reply app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt:91]: Ok, I've changed that everywhere.

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt (hunk 4 above)
Comment (jdolkowski) [reply app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt:52]: done

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (jdolkowski) [reply app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerFragment.kt:16]: changed to Dialog

**Review by sherrif10 — APPROVED** [review]
This looks good to me . cc @pwargulak.

File: app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt (hunk 3 above)
Comment (sherrif10) [reply app/src/androidTest/java/com/jnj/vaccinetracker/syncexecutionscript/DraftParticipantScript.kt:91]: You havent tackled this .

**Pull request merged**